### PR TITLE
added action bids, origin, strike event to pallet-society

### PR DESCRIPTION
**Pull request by joshua-mir**
TODO:

- [x] Kusama Companion PR.
- [ ] Tests for Actions.
- [ ] Ask folks if I have done this PR properly since I haven't done this in a while.

* Added a new BidType for members to execute arbitrary actions from a new Signed origin (note: NOT the society payout origin!) 
"Action" Bids have a minimum deposit tied to their size, where the deposit is the bid value, and reward their proposer with their deposit amount on success (along with execution of the call). Extended the bidding mechanism because I wanted this to play into the same "game".
* Added Strike event.
* Changed Strike logic and tests to only strike skeptics (cc @rrtti)
* Prevented Suspended members from being selected as skeptic
(cc @shawntabrizi @kianenigma)

### CODE REVIEW

**Comment by rrtti** [comment]
Is it be possible to make some changes, aiming to:

- don't penalize voters, only skeptics who have not voted. If non-skeptics vote, still loose part of the allocation if they are on "the wrong side": this way we just penalize skeptics and not normal voters, normal voters can increase the chances of a candidate to become a member, and they only lose if they re on the wrong side
- Reduce strikes to 3 if the first change is enacted.

File: bin/node/runtime/src/lib.rs [hunk 1]
```diff
@@ -112,8 +112,8 @@ pub const VERSION: RuntimeVersion = RuntimeVersion {
 	// and set impl_version to 0. If only runtime
 	// implementation changes and behavior does not, then leave spec_version as
 	// is and increment impl_version.
-	spec_version: 264,
-	impl_version: 1,
+	spec_version: 265,
```
Comment (kianenigma) [inline bin/node/runtime/src/lib.rs:115]: I don't think this really matters at this point :D but good that you maintain the practice

File: frame/society/src/lib.rs [hunk 2]
```diff
@@ -260,11 +266,11 @@ use sp_runtime::{Percent, ModuleId, RuntimeDebug,
 		TrailingZeroInput, CheckedSub
 	}
 };
-use frame_support::{decl_error, decl_module, decl_storage, decl_event, ensure, dispatch::DispatchResult};
+use frame_support::{Parameter, decl_error, decl_event, decl_module, decl_storage, dispatch::{DispatchResult, Dispatchable, GetDispatchInfo, PostDispatchInfo}, ensure};
```
Comment (kianenigma) [inline frame/society/src/lib.rs:269]:
```suggestion
use frame_support::{
	Parameter, decl_error, decl_event, decl_module, decl_storage, ensure,
	dispatch::{DispatchResult, Dispatchable, GetDispatchInfo, PostDispatchInfo},
};
```

File: frame/society/src/lib.rs [hunk 3]
```diff
@@ -413,12 +430,12 @@ decl_storage! {
 		pub Rules get(fn rules): Option<T::Hash>;
 
 		/// The current set of candidates; bidders that are attempting to become members.
-		pub Candidates get(fn candidates): Vec<Bid<T::AccountId, BalanceOf<T, I>>>;
+		pub Candidates get(fn candidates): Vec<Bid<T::AccountId, BalanceOf<T, I>, OpaqueCall>>;
```
Comment (kianenigma) [inline frame/society/src/lib.rs:433]:
```suggestion
		pub Candidates get(fn candidates): Vec<Bid<T::AccountId, BalanceOf<T, I>, <T as frame_system::Config>::Call>>;
```

Why not storing the call itself? it is encodable.

Comment (joshua-mir) [reply frame/society/src/lib.rs]: Honestly, I was having some obscure and hard to resolve errors with from/into when actually dispatching the Call - this was just my way to get the pallet to build and I haven't been able to get this working without the OpaqueCall type. - added this change but I'm reverting it now

**Comment by joshua-mir** [comment]
sorry accidentally push unrebased >.< fixed now!

File: frame/society/src/lib.rs (hunk 3 above)
Comment (kianenigma) [reply frame/society/src/lib.rs:436]: maybe I can also give it a go, if you have specific compilation issues?

Comment (joshua-mir) [reply frame/society/src/lib.rs:436]: would be welcome if you have the time - my philosophy here is that it works and isn't difficult to understand so I'm not going to mess with it too much 😅

**Comment by gnunicorn** [comment]
anyone working on this?

**Comment by joshua-mir** [comment]
@gnunicorn should I turn this into draft or close? I still want to get this in but it's not a high priority

**Comment by gnunicorn** [comment]
I'll close for now. Long-pending PRs are very hard to get merged due to the many changes on substrate. It is usually easier to just open a new PR against latest master whenever there's some focused time to work on things.

**Pull request closed without merging**